### DISCUSSION TITLE
Easier Windows Installation Options

### DISCUSSION THREAD

**Discussion by sheldonhull** [aquaproj/aqua#2165] [Deprecated (Proposal)] [answered]
### Overview

I'd like to contribute a few improvements to the windows tooling since I'm working with a lot of folks using windows now and want it to be smoother.

I would handle all the initial work for that to help the project, as I think it's something I might be able to uniquely help improve for you.

### Why is the proposal needed?

The following methods I'd like to evaluate but will need you to help, give me some limited rights to iron things out.

- A chocolatey package repo (wait before you do this to let me confirm).
- A scoop bucket repo so aqua published versions can push a new install to.
- Winget install
- PowerShell standalone install script.

**Comment by suzuki-shunsuke** [comment]
Thank you for your proposal!
It looks good to me.

I'm not familiar with chocolatey, scoop, and Winget, so I'm not sure what to do.
I'll take a look.

About PowerShell, we can add the script to https://github.com/aquaproj/aqua-installer .
We can also create a new repository for PowerShell, but I think it is better to unify the repository.

**Reply by suzuki-shunsuke** [reply]
- https://goreleaser.com/customization/winget/
- https://goreleaser.com/customization/scoop/
- https://goreleaser.com/customization/chocolatey/

**Comment by suzuki-shunsuke** [comment]
# PowerShell

About PowerShell, they can't execute [the shell script](https://github.com/aquaproj/aqua-installer/blob/main/aqua-installer) because bash isn't installed, right?

## Tasks

- [ ] Add a PowerShell script to aquaproj/aqua-installer
- [ ] Test the script in CI https://github.com/aquaproj/aqua-installer/tree/main/.github/workflows

**Reply by sheldonhull** [reply]
I'd be happy to do the installer script so basically a powershell user would just invoke via script. Agreed, that's great to put in the same repo. Testing via github action also makes sense.

No promises on timeframe yet, but I'm in!

**Comment by suzuki-shunsuke** [comment]
# Chocolatey

https://goreleaser.com/customization/chocolatey/
https://github.com/chocolatey/choco
https://docs.chocolatey.org/en-us/
https://docs.chocolatey.org/en-us/create/create-packages

## Tasks

- [ ] Create chocolatey repository
- [ ] Issue chocolatey API key
  - https://docs.chocolatey.org/en-us/create/commands/api-key
- [ ] Register API key to the repository secret of aquaproj/aqua
- [ ] Fix .goreleaser.yml of aquaproj/aqua

---

How can we create chocolatey repository and API key?

**Reply by sheldonhull** [reply]
The choco repository I'd be glad to either instruct step by step/script. If I confirm with you, you could add me as an owner temporarily to that single repository only so I could get everything running and hand off once it's done for your review. Like I said, not sure on timeframe but I'm pretty good about documenting so once I get it up you'd be able to take it over. Sound cool?

**Comment by suzuki-shunsuke** [comment]
# Winget

https://goreleaser.com/customization/winget/

> After releasing to GitHub, GitLab, or Gitea, GoReleaser can generate and publish a winget manifest and commit to a git repository, and PR it to winget-pkgs if instructed to.

https://learn.microsoft.com/en-us/windows/package-manager/winget/

https://learn.microsoft.com/en-us/windows/package-manager/package/

## Tasks

- [ ] Fork https://github.com/microsoft/winget-pkgs to https://github.com/aquaproj/winget-pkgs
- [ ] Add a machine user to `aquaproj/winget-pkgs`'s collaborator
- [ ] Add a machine user's GitHub Access Token to secrets
- [ ] Fix the release workflow
  - [ ]  Fix .goreleaser.yml of aquaproj/aqua
  - [ ] Fix .github/workflows/release.yaml

## What does the release workflow do?

- [ ] Generate winget manifests
- [ ] Create a remote branch and push winget manifests to `aquaproj/winget-pkgs`
- [ ] Send a pull request from `aquaproj/winget-pkgs` to https://github.com/microsoft/winget-pkgs

**Reply by sheldonhull** [reply]
- Looks like you fork the repo (see https://github.com/goreleaser/winget-pkgs) and then the forked repo is used for generating the pull request.
- We'd want to setup step prior to the goreleaser run that sync the fork so it's updated, goreleaser won't do that per their docs.

**Comment by suzuki-shunsuke** [comment]
# Scoop

https://goreleaser.com/customization/scoop/

> After releasing to GitHub, GitLab, or Gitea, GoReleaser can generate and publish a Scoop App Manifest into a repository that you have access to.

https://scoop.sh/
https://github.com/ScoopInstaller/Scoop

> A command-line installer for Windows

- Scoop App Manifest
  - https://github.com/ScoopInstaller/Scoop/wiki/App-Manifests

## Tasks

- [ ] Fix .goreleaser.yml of aquaproj/aqua

**Reply by sheldonhull** [reply]
This one is one of the easiest, I did this with a project already so if you create the equivalent repo of https://github.com/goreleaser/scoop-bucket, I can submit the scoop changes for goreleaser here and then it will publish to there.
Note, I had to disable branch protection when I did it, as goreleaser at the time wouldn't PR it, so I wanted it to just always create the new manifest without being blocked.

**Reply by suzuki-shunsuke** [reply]
📝 

https://github.com/ScoopInstaller/Scoop#known-application-buckets

aqua doesn't meet the criteria of `main` bucket.

https://github.com/ScoopInstaller/Scoop/wiki/Criteria-for-including-apps-in-the-main-bucket

> a reasonably well-known and widely used developer tool (e.g. if it's a GitHub project, it should have at least 500 stars and 150 forks)

So it seems there are two buckets.

- https://github.com/ScoopInstaller/Extras
- Create our own bucket e.g. `aquaproj/scoop-bucket`

Pros of Extras

👍 We don't need to manage a repository
👍 If users already use Extras, they don't have to install a bucket for aqua

Const of Extras

👎 We have to wait for review and merge of pull requests
👎 We can't control the bucket ourselves

**Reply by sheldonhull** [reply]
Yeah, just create the bucket. No need to be in the main distribution IMO. 
Just add as your own bucket option just as mentioned. I doubt the scoop discovery from main bucket is significant, it's more important just to have that install snippet in your docs directly.

**Comment by suzuki-shunsuke** [comment]
I'm not familiar with these tools, so I'm verifying them with my other tools.

- https://github.com/suzuki-shunsuke/circleci-config-merge/pull/514
- https://github.com/suzuki-shunsuke/goreleaser-windows-test
- https://github.com/mini-core/winget-pkgs/pulls
- https://github.com/suzuki-shunsuke/go-release-workflow/pull/43

**Reply by suzuki-shunsuke** [reply]
aqua uses the reusable workflow https://github.com/suzuki-shunsuke/go-release-workflow .

https://github.com/aquaproj/aqua/blob/main/.github/workflows/release.yaml

So we need to fix this workflow.

- https://github.com/suzuki-shunsuke/go-release-workflow/pull/43

**Comment by suzuki-shunsuke** [comment]
# Workaround: Install with Go

Go is required.

https://aquaproj.github.io/docs/install#go

```sh
go install github.com/aquaproj/aqua/v2/cmd/aqua@latest
```

Once you install aqua with Go, you can install pre built binaries by `aqua update-aqua`.

```sh
aqua update-aqua
```

**Comment by suzuki-shunsuke** [comment]
I got a Windows PC, so I can work on it.

**Comment by suzuki-shunsuke** [comment, marked as answer]
I created GitHub Issues to track the progress and situation of each task as GitHub Issue.

- https://github.com/aquaproj/aqua/issues/2400
- https://github.com/aquaproj/aqua/issues/2401
- https://github.com/aquaproj/aqua/issues/2402
- https://github.com/aquaproj/aqua/issues/2403

So I close this discussion.

**Comment by airtonix** [comment]
Comtrya has some prior art you can copy:

- on linux/mac: https://github.com/comtrya/get.comtrya.dev/blob/main/public/index.html
- on windows: https://github.com/comtrya/get.comtrya.dev/blob/main/public/ps.html


```
iwr "https://get.comtrya.dev/ps" -UseBasicParsing | iex
```

**Reply by airtonix** [reply]
see https://github.com/aquaproj/aqua-installer/issues/577
